SYCL/ESIMD/private_memory_access.cpp

```diff
@@ -0,0 +1,179 @@
+//==--------------- private_memory_access.cpp - DPC++ ESIMD on-device test -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
```

Comment:
Please note that @DenisBakhvalov  is enabling ESIMD tests on windows and level_zero.
Please make sure that you are aligned (I do expect that it should be supported).